File: 1207.unique-number-of-occurrences.cpp
/*
 * @lc app=leetcode id=1207 lang=cpp
 *
 * [1207] Unique Number of Occurrences
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool uniqueOccurrences(vector<int> &arr)
    {
        unordered_map<int, int> map;
        unordered_set<int> set;

        for (const int &num : arr)
        {
            if (map.find(num) != end(map))
            {
                map[num]++;
            }
            else
            {
                map[num] = 1;
            }
        }

        for (const pair<int, int> &key_value : map)
        {
            if (set.find(key_value.second) != end(set))
            {
                return false;
            }
            set.insert(key_value.second);
        }

        return true;
    }
};
// @lc code=end

File: 1207.unique-number-of-occurrences.py
#
# @lc app=leetcode id=1207 lang=python3
#
# [1207] Unique Number of Occurrences
#
from typing import *
# @lc code=start
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        map = {}
        
        for num in arr:
            map[num] = map.get(num, 0) + 1
        
        seen = set()
        
        for count in map.values():
            if count in seen:
                return False
            seen.add(count)
        
        return True
# @lc code=end

